src/test/java/assertions/PersonTest.java

```diff
@@ -1,42 +1,109 @@
 package assertions;
 
+import org.junit.Assert;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.ExpectedException;
 import sun.reflect.generics.reflectiveObjects.NotImplementedException;
 
 import java.time.LocalDate;
+import java.time.Period;
 
 /**
  * Created by ideadapt on 09.10.16.
  */
 public class PersonTest {
 
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
     // getFullName
 
     @Test
     public void getFullNameReturnsFirstnameSpaceLastname(){
-        // TODO implement
-        throw new NotImplementedException();
+
+        Person p = new Person("Hans", "Meier",LocalDate.now());
+
+        String fullname = p.getFullName();
+
+        Assert.assertEquals("Hans Meier", fullname);
+    }
+
+    @Test
+    public void getFullNameReturnsFirstnameSpaceLastnameInFirstnameMoreSpacesLastname(){
+        Person p = new Person(" Hans ", " Meier ", LocalDate.now());
+
+        String fullname = p.getFullName();
+
+        Assert.assertEquals("Hans Meier", fullname);
+    }
+
+    @Test
+    public void newPersonThrowIllegalArgumentExceptionFirstnameOrLastnameEmpty(){
+
+        thrown.expect(IllegalArgumentException.class);
+        thrown.expectMessage("not allowed empty String or null");
+
+        new Person("Hans", "  ", LocalDate.now());
```

Comment:
Die Frage ist hier was bedeutet "empty" :) Im Code testest du ja mit "inString.trim().isEmpty()", das heisst `"  "` wäre nicht direkt empty (erst nach einem trim).
Hier hättest du gut zwei Tests schreiben können, einer für IsEmpty, und einer für OnlyWhitespaces o.ä.

Im Testnamen gaukelst du vor, dass du FirstnameOrLastname empty testen wird, du testest aber nur den Lastname Fall. Hier hättest du auch gut zwei Tests schreiben können. 
(Or und And im Testnamen deuten meist darauf hin, dass man zuviel auf einmal testen will)